aggregator/counter_elem.gen.go

```diff
@@ -417,7 +420,9 @@ func (e *CounterElem) processValueWithAggregationLock(
 				value = res.Value
 			}
 		}
-		// Do we need to flush or forward NaNs?
+		if discardNaNValues && math.IsNaN(value) {
```

Comment:
nit: should this condition be the other way? ie we only need to check `discardNaNValues` if `value` is a NaN

Reply:
Yeah I imagine checking `discardNaNValues` is probably cheaper than checking if a value is NaN or not, and if so can short circuit early if the `discardNaNValues` is false.